pkg/network/http/ebpf_ssl.go

```diff
@@ -24,108 +19,79 @@ var sslProbes = []string{
+			unregisterCB: removeHooks(m, cryptoProbes),
+		},
+	)
 
-func createSSLPrograms(mainProgram *ebpfProgram, offsets []manager.ConstantEditor, libraries []so.Library) []subprogram {
-	var subprograms []subprogram
-	for i, lib := range libraries {
-		if !strings.Contains(lib.HostPath, "ssl") {
-			continue
-		}
+	watcher.Start()
+}
 
-		sslProg, err := newSSLProgram(mainProgram, offsets, lib.HostPath, i)
-		if err != nil {
-			log.Errorf("error creating SSL program for %s: %s", lib.HostPath, err)
-			continue
+func addHooks(m *manager.Manager, probes []string) func(string) error {
+	return func(libPath string) error {
+		uid := generateUID(libPath)
+		for _, sec := range probes {
+			p, found := m.GetProbe(manager.ProbeIdentificationPair{uid, sec})
+			if found {
+				if !p.IsRunning() {
+					err := p.Attach()
+					if err != nil {
+						return err
+					}
+				}
+
+				continue
+			}
+
+			newProbe := manager.Probe{
+				Section:    sec,
+				BinaryPath: libPath,
+				UID:        uid,
+			}
+
+			err := m.AddHook(baseUID, newProbe)
+			if err != nil {
+				return err
+			}
 		}
 
-		subprograms = append(subprograms, sslProg)
+		return nil
 	}
-
-	return subprograms
 }
 
-func newSSLProgram(mainProgram *ebpfProgram, offsets []manager.ConstantEditor, libPath string, i int) (*sslProgram, error) {
-	if libPath == "" {
-		return nil, fmt.Errorf("path to libssl not provided")
-	}
-
-	uid := strconv.Itoa(i)
-	var probes []*manager.Probe
-	for _, sec := range sslProbes {
-		probes = append(probes, &manager.Probe{
-			Section:    sec,
-			BinaryPath: libPath,
-			UID:        uid,
-		})
-	}
-
-	return &sslProgram{
-		mainProgram: mainProgram,
-		mgr:         &manager.Manager{Probes: probes},
-		offsets:     offsets,
-		libPath:     libPath,
-		uid:         uid,
-	}, nil
-}
+func removeHooks(m *manager.Manager, probes []string) func(string) error {
+	return func(libPath string) error {
+		uid := generateUID(libPath)
+		for _, sec := range probes {
+			p, found := m.GetProbe(manager.ProbeIdentificationPair{uid, sec})
+			if !found {
+				continue
+			}
+			p.Detach()
+		}
 
-func (p *sslProgram) Init() error {
-	var selectors []manager.ProbesSelector
-	for _, sec := range sslProbes {
-		selectors = append(selectors, &manager.ProbeSelector{
-			ProbeIdentificationPair: manager.ProbeIdentificationPair{
-				Section: sec,
-				UID:     p.uid,
-			},
-		})
+		return nil
 	}
-
-	log.Debugf("https (libssl) tracing enabled. lib=%s", p.libPath)
-	return p.mgr.InitWithOptions(
-		p.mainProgram.bytecode,
-		manager.Options{
-			RLimit: &unix.Rlimit{
-				Cur: math.MaxUint64,
-				Max: math.MaxUint64,
-			},
-			ActivatedProbes: selectors,
-			ConstantEditors: p.offsets,
-			MapEditors:      setupSharedMaps(p.mainProgram, sslMaps...),
-		},
-	)
-}
-
-func (p *sslProgram) Start() error {
-	return p.mgr.Start()
 }
 
-func (p *sslProgram) Close() error {
-	return p.mgr.Stop(manager.CleanAll)
+func generateUID(s string) string {
+	return s
```

Comment:
Can this be removed, since it just returns the string unmodified?

Reply:
ah, sorry. I added as a placeholder and ended up forgetting about it. I'll remove it